src/lib/shadow-shim-helper-rs/src/shim_shmem.rs

```diff
@@ -444,7 +444,7 @@ pub mod export {
     ) -> *mut ShimShmemHostLock {
         let host = unsafe { host.as_ref().unwrap() };
         let mut guard = host.protected().lock();
-        let lock = &mut *guard as *mut _;
+        let lock = std::ptr::from_mut(&mut *guard);
```

Comment:
maybe double-check this one 😵

Reply:
Yeah this one made me think for a minute :) But the nice thing is that it wouldn't compile if any of the types were incorrect. I'll add some type annotations to better show what's going on.